src/cysignals/signals.pyx

```diff
@@ -29,11 +29,35 @@ from cpython.ref cimport Py_XINCREF, Py_CLEAR
 from cpython.exc cimport (PyErr_Occurred, PyErr_NormalizeException,
         PyErr_Fetch, PyErr_Restore)
 from cpython.version cimport PY_MAJOR_VERSION
+from posix.time cimport clock_gettime, CLOCK_MONOTONIC
 
 cimport cython
 import sys
 from gc import collect
 
+
+cdef inline bint timespec_is_zero(timespec t):
+    return t.tv_sec == 0 and t.tv_nsec == 0
+
+
+cdef inline bint timespec_less(timespec a, timespec b):
+    return a.tv_sec < b.tv_sec or (a.tv_sec == b.tv_sec and a.tv_nsec < b.tv_nsec)
+
+
+cdef inline timespec timespec_add(timespec a, long long b):
+    cdef long long d = cython.cdiv(a.tv_nsec + b, 1000000000)
```

Comment:
Micro-performance optimization considerations (basically does not matter):

Replacing cdiv with Python division supposedly gives 35% slowdown https://github.com/cython/cython/wiki/enhancements-division#final-decision-and-status  , but it may be compensated by the check below anyway. In any case using Python division is a bit cleaner, but this should be best in terms of performance.

If `b` is known to be positive (which is true in our use case) the check below is unnecessary though.